geoengine/types.py

```diff
@@ -238,186 +396,297 @@ def spatial_reference(self) -> str:
     '''
-
-    __data_type: str
+    __spatial_bounds: Optional[BoundingBox2D]
+    __data_type: Literal['MultiPoint', 'MultiLineString', 'MultiPolygon']
     __columns: Dict[str, VectorColumnInfo]
 
-    def __init__(self, response: Dict[str, Any]) -> None:
-        '''Initialize a new `VectorResultDescriptor`'''
-        super().__init__(response['spatialReference'])
-        self.__data_type = response['dataType']
-        self.__columns = {name: VectorColumnInfo.from_response(info) for name, info in response['columns'].items()}
-
-    def __repr__(self) -> str:
-        '''Display representation of the vector result descriptor'''
-        r = ''
-        r += f'Data type:         {self.data_type}\n'
-        r += f'Spatial Reference: {self.spatial_reference}\n'
-
-        r += 'Columns:\n'
-        for column_name in self.columns:
-            column_info = self.columns[column_name]
-            r += f'  {column_name}:\n'
-            r += f'    Column Type: {column_info.data_type}\n'
-            r += f'    Measurement: {column_info.measurement}\n'
-
-        return r
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        spatial_reference: str,
+        data_type: Literal['MultiPoint', 'MultiLineString', 'MultiPolygon'],
+        columns: Dict[str, VectorColumnInfo],
+        time_bounds: Optional[TimeInterval] = None,
+        spatial_bounds: Optional[BoundingBox2D] = None,
+        spatial_resolution: Optional[SpatialResolution] = None
+    ) -> None:
+        ''' Initialize a vector result descriptor '''
+        super().__init__(spatial_reference, time_bounds, spatial_resolution)
+        self.__data_type = data_type
+        self.__columns = columns
+        self.__spatial_bounds = spatial_bounds
 
-    @classmethod
+    @staticmethod
+    def from_response_vector(response: api.VectorResultDescriptor) -> VectorResultDescriptor:
+        '''Parse a vector result descriptor from an http response'''
+        assert response['type'] == 'vector'  # TODO: throw exception
+
+        sref = response['spatialReference']
+        data_type = response['dataType']
+        columns = {name: VectorColumnInfo.from_response(info) for name, info in response['columns'].items()}
+
+        time_bounds = None
+        # FIXME: datetime can not represent our min max range
+        # if 'time' in response and response['time'] is not None:
+        #    time_bounds = TimeInterval.from_response(response['time'])
+        spatial_bounds = None
+        if 'bbox' in response and response['bbox'] is not None:
+            spatial_bounds = BoundingBox2D.from_response(response['bbox'])
+        spatial_resolution = None
+        if 'resolution' in response and response['resolution'] is not None:
+            spatial_resolution = SpatialResolution.from_response(response['resolution'])
+
+        return VectorResultDescriptor(sref, data_type, columns, time_bounds, spatial_bounds, spatial_resolution)
+
+    @ classmethod
```

Comment:
why are there all these whitespaces between @ and property introduced? We should get rid of them.

This usually happens when you have a faulty file that gets formatted.

Reply:
[10e097e](https://github.com/geo-engine/geoengine-python/pull/94/commits/10e097ec2354ba6bbbdac3a06039415da0741b59)